src/gc/proxy.d

```diff
@@ -318,6 +321,20 @@ extern (C)
         return proxy.gc_inFinalizer();
     }
 
+    void gc_lock() nothrow @nogc
+    {
+        if( proxy is null )
+            return _gc.lock();
+        return proxy.gc_lock();
+    }
+
+    void gc_unlock() nothrow @nogc
+    {
+        if( proxy is null )
+            return _gc.unlock();
+        return proxy.gc_unlock();
+    }
```

Comment:
> I thought this was used exclusively for connecting multiple Druntime instances (from multiple shared objects). Is this also used for third-party GC implementations?

IMO, the proxy never was good enough for its proposed use, i.e. dynamic linking against another runtie (because of other objects not being shared). 
It could be used to swap out the GC implementation if we wouldn't compile it together with the rest of druntime. Right now, inlining binds it to the current implementation.